index.js

```diff
@@ -267,14 +267,18 @@ let runInFiberContext = function (testInterfaceFnName, before, after, fnName) {
     let runHook = function (hookFn) {
         return origFn(function (done) {
             executeHooksWithArgs(before)
-                .then(() => new Promise((resolve) => {
+                .then(() => new Promise((resolve, reject) => {
                     return Fiber(() => {
-                        hookFn.call(this)
+                        try {
+                            hookFn.call(this)
+                        } catch (e) {
+                            reject(e)
+                        }
                         resolve()
                     }).run()
                 }))
                 .then(() => executeHooksWithArgs(after))
-                .then(() => done(), () => done())
+                .then(done, done)
```

Comment:
I wanted to ensure that:
1. errors in `beforeHook`/`afterHook` would still be exposed *somehow*, and
2. errors in `before`/`beforeEach`/etc. would be propagated, fail the test, have a decent stack trace, and appear in the BaseReporter stats so reporters can report the failure properly.

I think this works now.